documentation/release_notes.markdown

```diff
@@ -10,6 +10,42 @@ canonical: "/puppetdb/latest/release_notes.html"
 [upgrading]: ./api/query/v4/upgrading-from-v3.html
 [puppetdb-module]: https://forge.puppetlabs.com/puppetlabs/puppetdb
 
+3.0.2
+-----
+
+PuppetDB 3.0.2 is a bugfix release to address performance issues with the
+aggregate-event-count and fact-paths endpoints as well as memory starvation involving structured-facts.
+
+### Bug Fixes and Maintenance
+
+* Previously PuppetDB cached prepared statements for SQL queries. This caused
+  memory starvation involving structured-facts where large structured-facts would
+  would create a new prepared statement object each update. The objects would
+  accumulate due to the non-zero cache size. This release addresses this issue
+  two-fold, by setting the cache size to 0 and changing our structured-facts
+  update queries to be more efficient.
+  ([PDB-1721](https://tickets.puppetlabs.com/browse/PDB-1721))
+
+* Consumers of the aggregate-event-counts endpoint were experiencing slow response times
+  with a large number of nodes and reports in PuppetDB. These performance issues have
+  have been addressed in this release. One notable outcome from these fixes is that
+  the endpoint now no longer requires a query parameter.
+  ([PDB-1809](https://tickets.puppetlabs.com/browse/PDB-1809))
```

Comment:
Wonder if this might read a little easier as "Given a large number of nodes and reports, consumers of...".